src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -46,16 +47,39 @@ public function __construct(ControllerNameParser $parser, LoggerInterface $logge
     }
 
     /**
-     * Loads a resource.
-     *
-     * @param mixed  $resource A resource
-     * @param string $type     The resource type
-     *
-     * @return RouteCollection A RouteCollection instance
+     * {@inheritdoc}
      */
     public function load($resource, $type = null)
     {
-        $collection = parent::load($resource, $type);
+        if (false !== $this->loading) {
+            // This can happen if a fatal error occurs in parent::load().
+            // Here is the scenario:
+            // - while routes are being loaded by parent::load() below, a fatal error
+            //   occurs (e.g. parse error in a controller while loading annotations);
```

Comment:
People using the composant standalone will have the fallback exception handler, which works fine.
The pretty-shinny error page is a feature of the framework bundle. Others can take inspiration from it if they want :)